source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -486,26 +504,24 @@ absl::optional<QueuedChunkPtr> ChunkQueue::pop(bool undelivered_only) {
   }
   QueuedChunkPtr chunk = std::move(queue_.front());
   queue_.pop_front();
-  bytes_enqueued_ -= chunk->data.length();
+  bytes_enqueued_ -= chunk->buffer_length;
+
+  // Move the corresponding data out.
+  out_data.move(received_data, chunk->buffer_length);
   return chunk;
 }
 
-const QueuedChunk& ChunkQueue::consolidate(bool delivered) {
-  if (queue_.size() == 1) {
-    queue_.front()->delivered = delivered;
-    return *(queue_.front());
+const QueuedChunk& ChunkQueue::consolidate() {
+  if (queue_.size() > 1) {
```

Comment:
Consolidate never happens in STEAMED mode. It only happens in BUFFERED_PARTIAL mode.

In STREAMED mode,  delivered flag means the chunk data is sent to the ext_proc server, and waiting for the response.  Once the response for that chunk is received from the ext_proc server, it will be dequeued and mutated and injected back to the filter chain.  This maintains the streaming processing of the chunk data in STREAMED mode.